app/assets/javascripts/discourse/controllers/bookmark.js.es6

```diff
@@ -0,0 +1,224 @@
+import Controller from "@ember/controller";
+import ModalFunctionality from "discourse/mixins/modal-functionality";
+import discourseComputed from "discourse-common/utils/decorators";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import { htmlSafe } from "@ember/template";
+import { ajax } from "discourse/lib/ajax";
+
+const START_OF_DAY_HOUR = 8;
+const REMINDER_TYPES = {
+  AT_DESKTOP: "at_desktop",
+  LATER_TODAY: "later_today",
+  NEXT_BUSINESS_DAY: "next_business_day",
+  TOMORROW: "tomorrow",
+  NEXT_WEEK: "next_week",
+  NEXT_MONTH: "next_month",
+  CUSTOM: "custom"
+};
+
+export default Controller.extend(ModalFunctionality, {
+  loading: false,
+  errorMessage: null,
+  name: null,
+  selectedReminderType: null,
+  closeWithoutSaving: false,
+  saveButtonClicked: false,
+  onCloseWithoutSaving: null,
+
+  onShow() {
+    this.setProperties({
+      errorMessage: null,
+      loading: true,
+      name: null,
+      selectedReminderType: null,
+      closeWithoutSaving: false,
+      saveButtonClicked: false
+    });
+
+    this.set("loading", false);
```

Comment:
not sure I understand here:

we have `loading:true` and `loading:false` in the same runloop that feels wrong. Could you explain the reason please?

Reply:
@jjaffeux this was a goof, I didn't mean to leave that in there; was just boilerplate from other modal controllers.